uwblueprint/headsupguys
# [Module Builder] Multiple Choice & Multiselect

[[Module Builder] Multiple Choice/Multi select](https://www.notion.so/uwblueprintexecs/Module-Builder-Multiple-Choice-Multi-select-3294b253658c4981ba7ece96635386f2)

## Implementation description

- Created new section props for `multipleChoice` and `multiSelect` to store question + options
- Components for both section types, functionality to add options
- Sending data to Redux upon editing to be updated in database upon `Save`
- Created preview components that show up in `ModulePreview`, containing question and options as they are added
- Single and double column displays (test manually for now)

## Steps to test

1. Create a new module
3. Add a `Multiple Choice` or `Multi-Select` section - edit the question and add options
4. Save the module and ensure the section just added is populated with the correct data

## What should reviewers focus on?

- Functionality: you can edit the text fields and the data gets saved
- If any UI changes should be made according to [Figma](https://www.figma.com/file/GhDd9C0XsyVDgwL5zzvDvO/HUG-S21?node-id=3419%3A79493)
- Code organization

## Next steps

- Preview modifications (interact with radio/checkbox?)
- Change num columns

## Checklist
